posts/views.py

```diff
@@ -37,3 +37,92 @@ def put(self, request):
             serializer.save()
             return Response(serializer.data)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+
+
+class PostView(views.APIView):
+    #@method_decorator(login_required)
+    def post(self, request):
+        # print(request.data)
+        """
+        Create new categories
+        """
+        categories = request.data.get("categories")
+        if categories is not None:
+            # author has defined categories
+            for cat in categories:
+                # check if category exists
+                try:
+                    cat_obj = Category.objects.get(category=cat)
+                except Category.DoesNotExist:
+                    cat_obj = None
+
+                if cat_obj is None:
+                    Category.objects.create(category=cat)
+
+
+        serializer = PostSerializer(data=request.data, context={'create': False})
+        if serializer.is_valid():
+            serializer.save()
+            return Response(serializer.data, status=status.HTTP_201_CREATED)
+        print(serializer.errors)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
```

Comment:
Add todo for when visibility is done please. Also create an issue with these locations so we can easily add it when done